### PR TITLE
Validate client sync

### PR DESCRIPTION
**Pull request by keymane**
Depends on  https://github.com/OpenSRP/opensrp-client-core/pull/27

### CODE REVIEW

File: opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                    db.markEventValidationStatus(BaseRepository.TYPE_InValid, inValidClientId);
+                }
+
+                for (String clientId : clientIds) {
+                    db.markClientValidationStatus(BaseRepository.TYPE_Valid, clientId);
+                }
+            }
+
+            JSONArray inValidEvents = results.has("events") ? results.getJSONArray("events") : new JSONArray();
+            if (inValidEvents.length() > 0) {
```
Comment (pld) [inline opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java]: should be DRYd up line 100-110 duplicate lines 88-96, also don't need to check length

File: opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+
+        try {
+
+             int fetchLimit = FETCH_LIMIT;
+            EventClientRepository db = VaccinatorApplication.getInstance().eventClientRepository();
+
+            List<String> clientIds = db.getUnValidatedClientBaseEntityIds(fetchLimit);
```
Comment (pld) [inline opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java]: is the validation status of what's returned here unset?

File: opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java (hunk 1 above)
Comment (keymane) [reply opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java]: So one loop is for clients and the other one for events

File: opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java (hunk 2 above)
Comment (keymane) [reply opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java]: yea, unset or invalid

File: opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                Log.e(getClass().getName(), "Validation sync failed.");
+                return;
+            }
+
+            JSONObject results = new JSONObject(response.payload());
+
+            if (results.has(getString(R.string.clients_key))) {
+                JSONArray inValidClients = results.getJSONArray(getString(R.string.clients_key));
+
+                for (int i = 0; i < inValidClients.length(); i++) {
```
Comment (pld) [inline opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java]: Can lines 88-98 and 103-113 be DRYd up? They look that same to me

Comment (keymane) [reply opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java]: They are different, one loop for client documents and another for events

Comment (pld) [reply opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java]: Yep, but isn't everything else the same? Couldn't you write a helper function to remove the duplication

**Pull request closed without merging**